catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -727,7 +727,15 @@ public void setAttribute(Attribute attribute) {
             Serializable value = attribute.getValue();
             if (name != null) {
                 if (value != null) {
-                    map.put(name, attribute);
+                    Attribute tmpAttr = map.get(name);
+                    if (tmpAttr == null) {
+                        map.put(name, attribute);
+                    } else {
+                        List<Serializable> tmpAttrValues = tmpAttr.getValues();
+                        tmpAttrValues.addAll(attribute.getValues());
```

Comment:
This behavior is counterintuitive to the idiom of getters/setters. This effectively turns `setAttribute` into `setOrAppendAttribute`. I'd prefer a new method with a more descriptive name for this behavior.

You should also probably check if the `AttributeDescriptor` has `multivalued == true`. We need to start respecting the properties of `AttributeDescriptor`s more.